tools/codegen/dest/register_dispatch_key.py

```diff
@@ -103,109 +175,115 @@ def gen_structured(self, g: NativeFunctionsGroup) -> List[str]:
             self.target,
             self.selector,
             self.rocm,
+            self.cpp_namespace,
             g
         )
         return list(mapMaybe(structured_gen.gen_one, g.functions()))
 
-    @method_with_native_function
-    def gen_unstructured(self, f: NativeFunction) -> Optional[str]:
-        inplace_meta = False
-        if not self.backend_index.has_backend(f):
-            if (self.backend_index.dispatch_key == DispatchKey.Meta and
-                    f.func.kind() is SchemaKind.inplace and
-                    # Defer to composites for meta implementation
-                    not f.has_composite_kernel and
-                    # Inplace list operations are not supported
-                    len(f.func.returns) == 1):
-                inplace_meta = True
-            else:
+    def gen_unstructured(self, f: NativeFunction, *, is_generated_wrapper: bool = False) -> Optional[str]:
+        with native_function_manager(f):
```

Comment:
The `with native_function_manager(f)` makes the diff painful, but there aren't too many changes in this function. As per the comment [in the last PR](https://github.com/pytorch/pytorch/pull/56835#discussion_r620822511), I can't easily use a decorator since I have a defaultable argument

Reply:
It's fine, reviewers can ignore whitespace in GH UI